### DISCUSSION TITLE
Standalone Search tab

### DISCUSSION THREAD

**Discussion by heliole** [libre-tube/LibreTube#6840] [Ideas]
### Describe your suggested feature

The search button at the top of the screen is inconvenient to reach, especially on big phones. Since searching is what you do most of the time when you start up the app, it would be better if it had its own "Search" tab in the navbar (like Spotify and Google Play Store have).

It also adds 2 benefits:

1. Currently when you search for something, the app shows the results but keeps the last active tab before searching (like Library or Downloads) highlighted at the navbar - this makes the UI feel inconsistent because it’s not clear where you actually are in the app. For me it's intuitive for the search results to be a separate tab and show as an active tab.
2. The last performed search and the scroll position in the list would be stored in the search tab (but not between app restarts). Currently you can accidentally close it, and if you don't have search history turned on you have to re-type.

### **1st Approach (Better)**  

Keep the current search UI the same, just transfer it to a tab.

In app sessions, remember the last(current) performed search along with the search results and the scroll position. (or optionally maintain a local stack of all the performed searches in the current app session, but the scroll position of only the current search)

A tap on the search tab always focuses the search bar, even if coming from another tab. 
On tap, if there is an existing search in the search bar, auto highlight it (meaning it will get deleted once you start typing).

Until the user makes the first search, the search results are empty so it's ok to have a blank screen below the search bar at first (optionally with search icon illustration in the middle), which the user wont see anyway unless they unfocus the search bar before searching for anything.

A bonus UX addition in this approach would be to prevent the user to be able to go back before the first search (to the blank search screen) after they make the first search.

### **2nd Approach (Worse)**  

> Follow Standard UX Conventions (example Spotify/Google Play Store):
> 
> - When the search tab isn't active, a tap on it takes users to a dedicated search screen.  
> - If the search tab is already active, a tap on it activates the search field (the current screen that appears when you click on the search button)
> 
> The small problem with this approach is that most other apps have meaningful content to show in that dedicated search tab when the search field isn't active, but idk what would this app show:
> 1. Showing just a blank screen doesn’t make sense.
> 2. Showing cached results from the last search between app restarts is weird.
> 4. Showing watch history/home/trending would be too big of a change to existing user settings, and doesn't make much sense either.
> 
> If user already searched for something before in the app session and they are currently on the other tab, the click on the search tab would return them to that remembered scroll position of the search they last made. But I don't see a case where the user would want to go back to the same search after they switched to other tabs. More logical is if they click on the search, they want to search for something new and the search bar should appear first.

### Other details

I don't see this feature being a toggle option alongside the current default, because the underlying changes to the routing would be tiring to manage.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
I do really appreciate the details you've put into this issue.

However, as per my point of view, I don't feel like adding a dedicated search tab because
- There already are a lot of tabs in the bottom navigation bar, and adding yet another tab would make the bottom bar quite cluttered (especially since we have a navigation bar preference to enable some more tabs, and the maximum of bottom navigation bar items is 5 which we already reached when using that configuration option)
- As of now, most users are very much used to the search button in the top bar, and I want to avoid such drastic UX changes that would result in the necessity of user adaption to the new way of using the app

What comes to my mind is a floating action button (e.g. in the bottom right) that could be enabled in the settings and would be a quick action at the bottom (thus easy to reach) to open the search view, but it's indeed not an ideal solution. 

As you supposed, making such a search tab optional is definitely not a possible solution, as the back navigation logic would have to deal with two different search fragment positions then, which is never a good idea.

**Reply by heliole** [reply]
> As of now, most users are very much used to the search button in the top bar, and I want to avoid such drastic UX changes that would result in the necessity of user adaption to the new way of using the app

I get what you are saying. But you could also view it like this: A search tab is standard modern UX, and what any new or existing user would expect to see in a maintained app.
